knimin/lib/tests/test_data_access.py

```diff
@@ -1260,6 +1301,142 @@ def test_extract_sample_plates(self):
+        with self.assertRaisesRegexp(ValueError, "echo machine"):
+            db.normalize_shotgun_plate(cid, 'test', 'does not exist',
+                                       np.zeros((16, 24)), np.zeros((16, 24)))
+
+    def test_normalize_shotgun_plate(self):
+        before = datetime.datetime.now()
+        self._create_test_echo()
+        cid = self._create_test_shotgun_plate()
+        nid = db.normalize_shotgun_plate(cid, 'test', 'a valid echo name',
+                                         np.arange(384).reshape(16, 24),
+                                         np.arange(384).reshape(16, 24) * 10)
+        after = datetime.datetime.now()
+        exp_sample = np.arange(384).reshape(16, 24)
+        exp_water = np.arange(384).reshape(16, 24) * 10
+
+        ### many of these values are ignored at this time as dont know
+        ### where they get established
+        exp = {'created_on': datetime.date.today(),
+               'email': 'test',
+               'echo_id': 'ignored',  # this is assertable via inline sql
+               'lp_date': 'ignored',
+               'lp_email': 'ignored',
+               'mosquito': 'ignored',
+               'shotgun_plate_id': cid,
+               'shotgun_library_prep_kit_id': 'ignored',
+               'shotgun_adapter_aliquot_id': 'ignored',
+               'qpcr_date': 'ignored',
+               'qpcr_email': 'ignored',
+               'qpcr_std_ladder': 'ignored',
+               'qpcr_id': 'ignored',
+               'discarded': 'ignored',
+               'plate_normalization_water': exp_water,
+               'plate_normalization_sample': exp_sample}
+
+        obs = db.read_normalized_shotgun_plate(nid)
+
+        self.assertEqual(set(obs.keys()), set(exp.keys()))
+        self.assertEqual(obs['email'], exp['email'])
+        self.assertTrue(before <= obs['created_on'] <= after)
+        self.assertEqual(obs['shotgun_plate_id'], exp['shotgun_plate_id'])
+        npt.assert_equal(obs['plate_normalization_water'],
+                         exp['plate_normalization_water'])
+        npt.assert_equal(obs['plate_normalization_sample'],
+                         exp['plate_normalization_sample'])
+
+    def test_read_normalized_shotgun_plate_bad_id(self):
+        with self.assertRaises(ValueError):
+            db.read_normalized_shotgun_plate(99999999)
+
+    def test_delete_normalized_shotgun_plate_bad_id(self):
+        with self.assertRaises(ValueError):
+            db.delete_normalized_shotgun_plate(99999999)
+
+    def test_delete_normalized_shotgun_plate(self):
+        self._create_test_echo()
+        cid = self._create_test_shotgun_plate()
+        nid = db.normalize_shotgun_plate(cid, 'test', 'a valid echo name',
+                                         np.arange(384).reshape(16, 24),
+                                         np.arange(384).reshape(16, 24) * 10)
+        works = db.read_normalized_shotgun_plate(nid)
+        db.delete_normalized_shotgun_plate(nid)
+        try:
```

Comment:
I am bit unclear about the logic here. To my understanding, this plate should have been gone. So what's expected is an error (`pass`). What `else` will be? What if `doesnotwork` worked? Above all, why this test has to be written in this way, instead of the `assertRaisesRegexp` way?